packages/eas-cli/src/utils/queries.ts

```diff
@@ -0,0 +1,151 @@
+import { confirmAsync, selectAsync } from '../prompts';
+import uniqBy from './expodash/uniqBy';
+
+const fetchMoreValue = '_fetchMore';
+
+export type PaginatedQueryResponse<QueryReturnType extends Record<string, any>> = {
+  queryResponse: QueryReturnType[];
+  queryResponseRawLength: number;
+};
+
+type BasePaginatedQueryArgs<QueryReturnType extends Record<string, any>> = {
+  limit: number;
+  offset: number;
+  queryToPerform: (
+    limit: number,
+    offset: number
+  ) => Promise<PaginatedQueryResponse<QueryReturnType>>;
+};
+
+type PaginatedQueryWithConfirmPromptArgs<QueryReturnType extends Record<string, any>> =
+  BasePaginatedQueryArgs<QueryReturnType> & {
+    promptOptions: {
+      readonly title: string;
+      renderListItems: (currentPage: QueryReturnType[]) => void;
+    };
+  };
+
+type PaginatedQueryWithSelectPromptArgs<QueryReturnType extends Record<string, any>> =
+  BasePaginatedQueryArgs<QueryReturnType> & {
+    promptOptions: {
+      readonly title: string;
+      createDisplayTextForSelectionPromptListItem: (queryItem: QueryReturnType) => string;
+      getIdentifierForQueryItem: (queryItem: QueryReturnType) => string;
+    };
+  };
+
+export async function paginatedQueryWithConfirmPromptAsync<
+  QueryReturnType extends Record<string, any>
+>({
+  limit,
+  offset,
+  queryToPerform,
+  promptOptions,
+}: PaginatedQueryWithConfirmPromptArgs<QueryReturnType>): Promise<void> {
+  return await _paginatedQueryWithConfirmPromptInternalAsync(
+    { limit, offset, queryToPerform, promptOptions },
+    []
+  );
+}
+
+// eslint-disable-next-line no-underscore-dangle
+async function _paginatedQueryWithConfirmPromptInternalAsync<
+  QueryReturnType extends Record<string, any>
+>(
+  {
+    limit,
+    offset,
+    queryToPerform,
+    promptOptions,
+  }: PaginatedQueryWithConfirmPromptArgs<QueryReturnType>,
+  accumulator: QueryReturnType[]
+): Promise<void> {
```

Comment:
```suggestion
export async function paginatedQueryWithConfirmPromptAsync<
  QueryReturnType extends Record<string, any>
>(
  {
    limit,
    offset,
    queryToPerform,
    promptOptions,
  }: PaginatedQueryWithConfirmPromptArgs<QueryReturnType>,
  accumulator?: QueryReturnType[] = []
): Promise<void> {
```

My only suggestion was about mutating `accumulator`. You applied this suggestion by introducing `newAccumulator`.

Reply:
yup. while I was thinking about the accumulator I realized it should just not be exposed at all. I want to explicitly disallow clients from providing an accumulator, it's purely there to support recursion.